tools/migration/cmd/go-filecoin-migrate.go

```diff
@@ -0,0 +1,105 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/filecoin-project/go-filecoin/tools/migration/internal"
+	"github.com/filecoin-project/go-filecoin/tools/migration/migrate_1-to-2"
+)
+
+type Migrator interface {
+	Migrate(oldRepo, newRepo *os.File) error
+	Describe()
+	Validate(oldRepo, newRepo *os.File) error
+}
+
+// Run runs the given migration command
+func Run(cmd string, verbose bool) error {
+	var err error
+	miglog, err := makeMigl()
+	logpath := "/tmp/migration_log.txt"
+
+	oldRepo, err := getOldRepo()
+	if err != nil {
+		miglog.Error(err)
+		return err
+	}
+
+	// open repodir read-only
+	// create new repodir & open read-write
+	// TODO: decide where to put the temp new repo.
+	newRepo, err := os.Create("/tmp/newRepo")
+	if err != nil {
+		miglog.Error(err)
+		return err
+	}
+
+	var migrator Migrator
+	if err != nil {
+		miglog.Error(err)
+		return err
+	}
+	// TODO see Issue #2585
```

Comment:
Please give a one-sentence summary of the TODO here, which I guess is to inspect the repo to determine its version. I don't think we should land without that bit of logic being implemented.